packages/itinerary-body/src/types.ts

```diff
@@ -0,0 +1,242 @@
+import { FunctionComponent, ReactElement } from "react";
+
+import {
+  Config,
+  Fare,
+  Itinerary,
+  Leg,
+  Place,
+  TimeOptions,
+  TransitOperator
+} from "@opentripplanner/types";
+
+export type LegIconComponent = FunctionComponent<{
+  leg: Leg;
+  title?: string;
+  width?: string;
+}>;
+
+export interface RouteDescriptionProps {
+  /** Contains details about leg object that is being displayed */
+  leg: Leg;
+  /** A component class that is used to render icons for legs of an itinerary */
+  LegIcon: LegIconComponent;
+  /** The transit operator associated with the route if available */
+  transitOperator: TransitOperator;
+}
+
+export type GradationMap = Record<
+  number,
+  { color: string; icon?: ReactElement; text?: string }
+>;
+
+export type ToRouteAbbreviationFunction = (route: string | number) => string;
+
+export interface TimeColumnContentProps {
+  /** whether this place row represents the destination */
+  isDestination: boolean;
+  /** Contains details about leg object that is being displayed */
+  leg: Leg;
+}
+
+/**
+ * Shared props for various components that render leg data.
+ */
+export interface LegSharedProps {
```

Comment:
Could this extend TimeColumnContentProps? It's not semantically sound but might help reduce line count if that's something we're worried about. There's a few fields that are very repeated

Reply:
See if https://github.com/opentripplanner/otp-ui/pull/357/commits/34ff0279e9ba5e10d012a29d7b79933f15c71abb makes things better or worse...

Reply:
I think I prefer it, but I'll leave this unresolved for the second reviewer to take a look at it.